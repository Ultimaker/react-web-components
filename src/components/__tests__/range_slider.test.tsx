// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import RangeSlider from '../range_slider';

describe('The Range Slider component', () => {
    it('should render', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<RangeSlider onChange={onChange} value={0.5} />);
        expect(wrapper).toMatchSnapshot();
        expect(onChange).not.toHaveBeenCalled();
    });

    it('should parse the callback', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<RangeSlider onChange={onChange} value={0.5} />);
        const inputChanged = wrapper.find('input').prop('onChange') as any;
        inputChanged({ target: { value: '0.9' } });
        expect(onChange).toHaveBeenCalledWith(0.9);
    });
});
