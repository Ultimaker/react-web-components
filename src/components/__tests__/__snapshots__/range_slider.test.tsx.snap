// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Range Slider component should render 1`] = `
<div
  className="range-slider"
>
  <input
    defaultValue="0.5"
    max={1}
    min={0}
    name="scale"
    onChange={[Function]}
    step={0.1}
    type="range"
  />
</div>
`;
